src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala

```diff
@@ -505,21 +524,21 @@ trait V2Formats {
       (__ \ "maxLaunchDelaySeconds").readNullable[Long]
       .withDefault(AppDefinition.DefaultMaxLaunchDelay.toSeconds).asSeconds ~
       (__ \ "container").readNullable[Container] ~
-      (__ \ "healthChecks").readNullable[Set[HealthCheck]].withDefault(AppDefinition.DefaultHealthChecks) ~
-      (__ \ "dependencies").readNullable[Set[PathId]].withDefault(AppDefinition.DefaultDependencies)
+      (__ \ "healthChecks").readNullable[Set[HealthCheck]].withDefault(AppDefinition.DefaultHealthChecks)
     )((
-        id, cmd, args, maybeString, env, instances, cpus, mem, disk, executor, constraints, uris, storeUrls,
-        requirePorts, backoff, backoffFactor, maxLaunchDelay, container, checks, dependencies
+
+        id, cmd, args, maybeString, env, instances, cpus, mem, disk, executor, constraints, uris, fetch, storeUrls,
+        requirePorts, backoff, backoffFactor, maxLaunchDelay, container, checks
       ) => AppDefinition(
         id = id, cmd = cmd, args = args, user = maybeString, env = env, instances = instances, cpus = cpus,
-        mem = mem, disk = disk, executor = executor, constraints = constraints, uris = uris,
-        storeUrls = storeUrls, ports = Nil,
-        requirePorts = requirePorts, backoff = backoff,
-        backoffFactor = backoffFactor, maxLaunchDelay = maxLaunchDelay, container = container, healthChecks = checks,
-        dependencies = dependencies
+        mem = mem, disk = disk, executor = executor, constraints = constraints,
+        fetch = mergeFetchWithUris(fetch, uris), storeUrls = storeUrls, ports = Nil, requirePorts = requirePorts,
```

Comment:
Yes it does but I don't think it's really a problem. Mesos handles duplicated fetch uris without a problem